crates/core/src/kernel/transaction/mod.rs

```diff
@@ -967,4 +1026,58 @@ mod tests {
             .await
             .unwrap();
     }
+
+    #[test]
+    fn test_commit_with_retries_tracing_span() {
+        let span = tracing::info_span!(
+            "commit_with_retries",
+            base_version = 5,
+            max_retries = 10,
+            attempt = tracing::field::Empty,
+            target_version = tracing::field::Empty,
+            conflicts_checked = 0
+        );
+
+        let metadata = span.metadata().expect("span should have metadata");
+        assert_eq!(metadata.name(), "commit_with_retries");
+        assert_eq!(metadata.level(), &tracing::Level::INFO);
+        assert!(metadata.is_span());
+
+        span.record("attempt", 1);
+        span.record("target_version", 6);
+        span.record("conflicts_checked", 2);
+    }
+
+    #[test]
+    fn test_commit_tracing_log_events() {
+        tracing::debug!("committing initial table version 0");
+        tracing::info!(
+            version = 6,
+            num_retries = 1,
+            "transaction committed successfully"
+        );
+        tracing::warn!(
+            base_version = 5,
+            latest_version = 7,
+            "table updated during transaction"
+        );
+        tracing::error!(version = 8, "commit failed, aborting");
+        tracing::debug!(conflicts_checked = 3, "all conflicts resolved");
+    }
```

Comment:
This test also doesn't seem to actually do anything :upside_down_face:

Reply:
I'm working on it... I removed some examples where I tested the execution.
I wanted to see the test coverage results first.